File: ui/src/types/mm.ts
import type { LogEntry } from './shared'

export interface ActiveQuote {
  market_id: string
  side: string
  price: number
  size: number
  order_id: string
}

export interface InventoryEntry {
  market_id: string
  net_yes: number
  realized_pnl: number
  volume: number
}

export interface MmState {
  active_quotes: ActiveQuote[]
  inventory: InventoryEntry[]
  total_realized_pnl: number
  fill_count: number
  running: boolean
  logs: LogEntry[]
}

File: src/commands/dto/mm.rs
//! Market making tab DTOs.

use serde::Serialize;

#[derive(Serialize, Clone)]
pub struct ActiveQuoteDto {
    pub market_id: String,
    pub side: String,
    pub price: f64,
    pub size: f64,
    pub order_id: String,
}

#[derive(Serialize, Clone)]
pub struct InventoryEntryDto {
    pub market_id: String,
    pub net_yes: f64,
    pub realized_pnl: f64,
    pub volume: f64,
}

#[derive(Serialize, Clone)]
pub struct MmStateDto {
    pub active_quotes: Vec<ActiveQuoteDto>,
    pub inventory: Vec<InventoryEntryDto>,
    pub total_realized_pnl: f64,
    pub fill_count: u32,
    pub running: bool,
    pub logs: Vec<super::shared::LogEntryDto>,
}
